### PR TITLE
Fix: Let versioned  controller uses a separate election locker in one cluster.

### PR DESCRIPTION
**Pull request by leejanee**
Signed-off-by: Jian.Li <lj176172@alibaba-inc.com>

### Description of your changes

Fixes #

 Let versioned  controller uses a separate election locker in one cluster.
I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [x] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

**Review by hongchaodeng — APPROVED** [review]

File: pkg/utils/util/factory.go [hunk 1]
```diff
@@ -167,3 +169,8 @@ func computeDiscoverCacheDir(parentDir, host string) string {
 	safeHost := overlyCautiousIllegalFileCharacters.ReplaceAllString(schemelessHost, "_")
 	return filepath.Join(parentDir, safeHost)
 }
+
+// GenerateLeaderElectionID returns the Leader Election ID.
+func GenerateLeaderElectionID(name string, versionedDeploy bool) string {
+	return name + "-" + strings.ToLower(strings.ReplaceAll(version.VelaVersion, ".", "z"))
```
Comment (wonderflow) [inline pkg/utils/util/factory.go:175]:
```suggestion
	return name + "-" + strings.ToLower(strings.ReplaceAll(version.VelaVersion, ".", "-"))
```

File: pkg/utils/util/factory_test.go [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import (
+	"testing"
+
+	"github.com/oam-dev/kubevela/version"
+)
+
+func TestGenerateLeaderElectionID(t *testing.T) {
+	version.VelaVersion = "v10.13.0"
+	if id := GenerateLeaderElectionID("kubevela", true); id != "kubevela-v10z13z0" {
+		t.Errorf("id is not as expected(%s != kubevela-v10z13z0)", id)
```
Comment (wonderflow) [inline pkg/utils/util/factory_test.go:12]:
```suggestion
		t.Errorf("id is not as expected(%s != kubevela-v10-13-0)", id)
```

File: pkg/utils/util/factory.go [hunk 3]
```diff
@@ -23,6 +23,8 @@ import (
 	"strings"
 	"time"
 
+	"github.com/oam-dev/kubevela/version"
+
```
Comment (wonderflow) [inline pkg/utils/util/factory.go:27]: format

File: cmd/core/main.go [hunk 4]
```diff
@@ -30,6 +30,8 @@ import (
 	"strings"
 	"time"
 
+	"github.com/oam-dev/kubevela/pkg/utils/util"
+
```
Comment (wonderflow) [inline cmd/core/main.go:34]: format

File: pkg/utils/util/factory.go (hunk 1 above)
Comment (wonderflow) [inline pkg/utils/util/factory.go:175]:
```
if versionedDeploy{
  return name+"-"+...
}
return name

```

File: pkg/utils/util/factory_test.go [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+import (
+	"testing"
+
+	"github.com/oam-dev/kubevela/version"
+)
+
+func TestGenerateLeaderElectionID(t *testing.T) {
+	version.VelaVersion = "v10.13.0"
+	if id := GenerateLeaderElectionID("kubevela", true); id != "kubevela-v10z13z0" {
```
Comment (wonderflow) [inline pkg/utils/util/factory_test.go:11]: add a test about false case

**Review by wonderflow — CHANGES_REQUESTED** [review]
please make sure it's compatible

**Review by wonderflow — APPROVED** [review]

**Pull request merged**